banzai/qc/header_checker.py

```diff
@@ -45,144 +45,99 @@ def do_stage(self, images):
-                header_value).__name__ + ' in place of ' + self.header_expected_format[
-                    keyword].__name__)
-
-            self.logger.error(sentence, extra=logging_tags)
-
-    def check_header_na(self, keywords, image):
-        """ Logs an error if the keyword is 'N/A' instead of the
-            expected type in the image header.
-
-        Parameters
-        ----------
-        keyword : str
-                  the keyword of interest
-        image : object
-                a  banzais.image.Image object.
-
-        """
-        logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
-        na_keywords = []
-        for keyword in keywords:
-            try:
-                header_value = image.header[keyword]
-                if isinstance(header_value, str):
-                    if 'N/A' in header_value:
-                        sentence = 'The header key ' + keyword + ' got the unexpected value : N/A'
-                        self.logger.error(sentence, extra=logging_tags)
-                        na_keywords.append(keyword)
-            except:
-                pass
-        any_keywords_na = True if len(na_keywords) else False
-        qc_results = {"header.keywords.na.failed": any_keywords_na}
-        if any_keywords_na:
+        if are_keywords_na:
             qc_results["header.keywords.na.names"] = na_keywords
         self.save_qc_results(qc_results, image)
+        return missing_keywords + na_keywords
 
-    def check_ra_range(self, image):
+    def check_ra_range(self, image, bad_keywords=None):
         """ Logs an error if the keyword right_ascension is not inside
             the expected range (0<ra<360 degrees) in the image header.
 
         Parameters
         ----------
         image : object
                 a  banzais.image.Image object.
+        bad_keywords: list
+                a list of any keywords that are missing or NA
 
         """
-        ra_value = image.header['CRVAL1']
-        logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
-        try:
-            bad_ra_value = (ra_value > self.RA_MAX) | (ra_value < self.RA_MIN)
-            if bad_ra_value:
+        if bad_keywords is None:
+            bad_keywords = []
+        if 'CRVAL1' not in bad_keywords:
+            logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
+            ra_value = image.header['CRVAL1']
+            is_bad_ra_value = (ra_value > self.RA_MAX) | (ra_value < self.RA_MIN)
+            if is_bad_ra_value:
                 sentence = 'The header CRVAL1 key got the unexpected value : ' + str(ra_value)
                 self.logger.error(sentence, extra=logging_tags)
-            self.save_qc_results({"header.ra.failed": bad_ra_value,
+            self.save_qc_results({"header.ra.failed": is_bad_ra_value,
                                   "header.ra.value": ra_value}, image)
```

Comment:
Should you save the range used? Or is 0 to 360 obvious?

Reply:
Partly because it's obvious, but also partly because it's set by the definition of RA and will never change, so there is no need to keep track of it.